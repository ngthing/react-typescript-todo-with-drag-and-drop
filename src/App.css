:root {
    --background: #f9f2f2;
    --textColor: rgb(43, 39, 39);
    --checkboxBorder: #3d9863;
    --checkboxColor: #6ec26eb5;
    --white: #ffffff;
    --todoBorder: #c0bbbb8f;
}

body {
    background-color: var(--background);
    color: var(--textColor);
    min-height: 100vh;
}

.todo-list-name {
    padding-top: 15vh;
    padding-bottom: 5vh;
}

.todo-list-name input {
    font-size: 3rem;
    text-align: center;
    cursor: pointer;
}

.header {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.todo-list-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    /* color: white; */
    padding-bottom: 20px;
}

.todo {
    display: flex;
    align-items: center;
    margin: 0.25rem 0;
    border: solid var(--todoBorder);
    padding: 2px 10px;
    border-radius: 16px;
    min-width: 50%;
}

.todo-is-completed .checkbox {
    /* color: #433c3c; */
    background: var(--checkboxColor);
}

.todo-is-completed input {
    text-decoration: line-through;
    text-decoration-color: #ada1a1;
    text-decoration-thickness: 2px;
}

.checkbox {
    width: 25px;
    height: 22px;
    border-radius: 50%;
    margin: 0 10px;
    cursor: pointer;
    font-size: 13px;
    display: flex;
    justify-content: center;
    align-items: center;
    transition: background-color .2s ease-in-out;
    border: 1px solid var(--checkboxBorder);
}

.checkbox:hover {
    background: var(--checkboxColor);
    border: 1px solid rgba(255, 255, 255, 0);
}

input {
    border: none;
    background: transparent;
    /* color: white; */
    font-size: 1.4rem;
    outline: none;
    width: 100%;
}

.btn-action {
    border-color: transparent;
    background: #c6b9b9;
    border-radius: 45%;
    color: #2f2b2b;
    cursor: pointer;
    width: 30px;
    height: 25px;
}

.btn-action:hover {
    background: #ece2e2;
}

.todo-btns-action {
    display: flex;
    justify-content: center;
}

.todo-index {
    cursor: pointer;
}

;