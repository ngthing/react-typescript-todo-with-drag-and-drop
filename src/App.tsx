import React, { useEffect, useState } from 'react';
import { TodoList } from './TodoList';
import { sampleTodos } from './data';

import './App.css';


function App() {
    const storedTodoName = localStorage.getItem('todoName') ?? 'Toododo';
    const storedTodos = (JSON.parse(localStorage.getItem('todos') ?? '[]')) as Todo[];

    const [todoName, setTodoName] = useState(storedTodoName.length ? storedTodoName : 'Toododo')
    const [todos, setTodos] = useState(storedTodos.length ? storedTodos : sampleTodos);

    useEffect(() => {
        localStorage.setItem('todoName', todoName)
    }, [todoName]);

    useEffect(() => {
        localStorage.setItem('todos', JSON.stringify(todos))
    }, [todos])

    return (
        <div className="app">
            <TodoList todos={todos} name={todoName} onNameChange={setTodoName} setTodos={setTodos} />
        </div >
    );
}

export default App;