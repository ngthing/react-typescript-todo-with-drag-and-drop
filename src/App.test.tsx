import { render, screen } from '@testing-library/react';
import App from './App';


describe('App', () => {
    it('renders App component and has a default Todo list', () => {
        render(<App />);
        expect(screen.getByLabelText('todoName')).toBeInTheDocument();
        expect(screen.getByLabelText('add-todo')).toBeInTheDocument();
        expect(screen.getByLabelText('delete-todo-list')).toBeInTheDocument();
        expect(screen.getByLabelText('check-all-todo')).toBeInTheDocument();
        expect(screen.getByLabelText('uncheck-all-todo')).toBeInTheDocument();
    });
});

