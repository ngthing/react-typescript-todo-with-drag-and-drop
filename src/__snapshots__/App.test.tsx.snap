// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg app css-1oqqzyl-MuiContainer-root"
  data-width={1024}
>
  <div
    className="main-container MuiBox-root css-0"
  >
    <div
      className="sc-hKwCKL guGzDs"
    >
      <input
        aria-label="todoName"
        className="sc-eCImbG jzTxPy"
        id="todoName"
        onChange={[Function]}
        type="text"
        value="Toododo"
      />
    </div>
    <div
      className="todo-list-stats"
    >
      <span>
        Completed 
        1
         / 
        4
      </span>
      <span
        className="completedCheck"
      >
        âœ”
      </span>
    </div>
    <div
      className="sc-bdvuGq eDNor"
    >
      <button
        aria-label="check-all-todo"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1m81jh1-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="DoneAllIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m18 7-1.41-1.41-6.34 6.34 1.41 1.41L18 7zm4.24-1.41L11.66 16.17 7.48 12l-1.41 1.41L11.66 19l12-12-1.42-1.41zM.41 13.41 6 19l1.41-1.41L1.83 12 .41 13.41z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="uncheck-all-todo"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1m81jh1-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="UnpublishedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21.19 21.19 2.81 2.81 1.39 4.22l2.27 2.27C2.61 8.07 2 9.96 2 12c0 5.52 4.48 10 10 10 2.04 0 3.93-.61 5.51-1.66l2.27 2.27 1.41-1.42zm-10.6-4.59-4.24-4.24 1.41-1.41 2.83 2.83.18-.18 1.41 1.41-1.59 1.59zm3-5.84-7.1-7.1C8.07 2.61 9.96 2 12 2c5.52 0 10 4.48 10 10 0 2.04-.61 3.93-1.66 5.51L15 12.17l2.65-2.65-1.41-1.41-2.65 2.65z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="copy-list"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1m81jh1-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ContentCopyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="add-todo"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1m81jh1-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="delete-todo-list"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ClearIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <footer
    className="sc-gsDLol ixyHkR"
  >
    <div
      className="sc-dkPuFg evkqoj"
    >
      ðŸŒ± Inspired by the
       
      <a
        href="https://github.com/ngthing/toododo"
        target="blank"
      >
        Internet
      </a>
      . Made by
       
      <a
        href="https://justthinguyen.com"
        target="blank"
      >
        Thi Nguyen
      </a>
      . Thanks for stopping by and have a beautiful day! ðŸŒ±
    </div>
  </footer>
</div>
`;
